src/CombFilter/CombFilter.cpp

```
@@ -0,0 +1,157 @@
+    for (int i = 0; i < iNumChannels; i++){
+        pCRingBuff[i] = new CRingBuffer<float>(kBlockSize);
+    }
+    
+    
+    m_bIsInitialized = true;
+    
+    return Error_t::kNoError;
+}
+
+Error_t CCombFilterBase::reset ()
+{
+    //resets the internal variables (requires new call of init)
+    
+    
+    for(int i = 0; i < m_iNumberOfChannels; i++){
+        delete pCRingBuff[i];
+    }
+    
+    m_fSampleRate = 0;
+    m_iNumberOfChannels = 0;
+    m_ParamGain = 0.;
+    m_ParamDelay = 0.;
+    
+    m_bIsInitialized = false;
+    
+    
+    return Error_t::kNoError;
+}
+
+Error_t CCombFilterBase::process (float **ppfInputBuffer, float **ppfOutputBuffer, int iNumberOfFrames)
+{
+    
+    for (int i = 0; i < iNumberOfFrames; i++){
+        for (int c = 0; c < m_iNumberOfChannels; c++)
+        {
+
+            ppfOutputBuffer[c][i] = pCRingBuff[c]->getPostInc();
+            pCRingBuff[c]->putPostInc(ppfInputBuffer[c][i]);
+            
+        }
+        
+    }
+    
+    return Error_t::kNoError;
+}
+
+Error_t CCombFilterBase::setParam (int iParam, float fParamValue)
+{
+    FilterParam_t eParam = (FilterParam_t)iParam;
+        
+    switch(eParam){
+        case FilterParam_t::kParamGain:
+            if (fParamValue < -1. || fParamValue > 1.){
+                return Error_t::kFunctionInvalidArgsError;
+            }
+            m_ParamGain = fParamValue;
+            
+        case FilterParam_t::kParamDelay:
+            if (fParamValue < 0. || fParamValue >= m_fMaxDelayLengthInS){
+                return Error_t::kFunctionInvalidArgsError;
+            }
+            
+            m_ParamDelay = fParamValue;
+            m_iDelayinSamples = (int) (m_ParamDelay * m_fSampleRate);
+            for (int c = 0; c < m_iNumberOfChannels; c++){
+                pCRingBuff[c]->setReadPtr(m_iDelayinSamples);
+            }
+            
+        case kNumFilterParams:kNumFilterParams:
+            break;
```

Maybe this should return an error, since it's not a valid parameter? (Then, since you're using a switch, this could just be the `default:` case.)